File: exercism/haskell/raindrops/src/Raindrops.hs
module Raindrops (convert) where

import Data.Bool (bool)

convert :: Int -> String
convert n = bool s (show n) (null s)
 where
  s = concatMap (\(k, p) -> bool "" p (0 == n `mod` k))
                [(3, "Pling"), (5, "Plang"), (7, "Plong")]

File: exercism/ruby/raindrops/raindrops.rb
module Raindrops
  SOUNDS = {3 => :Pling, 5 => :Plang, 7 => :Plong}
  def self.convert(i)
    s = SOUNDS.map { |n, p| p if (i % n).zero? }.join
    s.empty? ? i.to_s : s
  end
end

module BookKeeping
  VERSION = 3
end
